### ISSUE TITLE
Conversations

### ISSUE THREAD

**Issue by rheadkid** [rheadkid/Quest-Group_Project#12]
Will SPEAK trigger a menu? Or will the player have to figure out what to ask/tell about/for?

NOTE: In the game I'm currently porting from Inform to Quest, I'm going all out with the ```displayverbs``` and ```topics``` attributes. There's a commands pane, and all sorts of bells and whistles, like commands that hide/show the side panes and others that change the CSS from my custom scheme to the Quest default scheme. I'm designing it so you can click the whole way through OR lose the clicking options and let your fingers do the adventuring. (I presume most player's who are clicking rather than parsing are playing on mobile devices, hence are using their thumbs to play the game...)

---
Sorry, I digress quite frequently.

---
Having an interface outside of the parser is extra work to design, but that's really outside of the point here. (NOTE TO SELF: Edit this out and start a separate DISPLAY issue.)

We can still have a list of topics display when speaking/talking to an NPC with no interface besides the parser (which I believe is the route we'll be taking).

This is probably wise to set up, Then each author can control how the command is scripted out in their respective universes.

**Comment by DarkLizerd** [comment]
Menu driven is the easiest to control, and no guessing the next word to continue the conversation. 
To make this feel like a consistent package, we all need the same interface in all parts of the worlds.
> ask bob about sword
is hard to work out all the different ways this could be said... but
>talk bob    or talk to bob
Bob replies "Hi, {player.name}, how can I help you?
Then a list of topics is displayed with links, would prevent guessing.
And, if the player was told to 'ask Bob about his sword' then 'sword' is added to the list.
For simplicity, I would go for the menu driven conversations.

**Comment by rheadkid** [comment]
I **_think_** we're going pure parser, but that was just what @ThePix  and I discussing it...

I don't know if that makes it easier to put all the pieces together or not...

I'm pretty much done with my one-room sample template. My NPC prattles on when you SPEAK to him, but you can ask him about (pretty much) each thing he mentions during the conversation. (I'll add in responses for any topic I missed in the real sample.)

SPEAK TO FARMER
Blah, blah... hoes.

ASK HIM ABOUT HOES
Blah, blah, topic1, blah, blah, topic2, here, have a hoe.

Then you ask about topic1|topic2, which open the door up even further, and so on...

It's how I've rolled for 25 years, but I did pretty much stick to Infocom games. (USE doesn't do jack shit in Infocom games either, so I'll be implementing every other noun+verb combinations I can think of that will call the USE scripts. The USE will be set up with the main script, of course.)

If it's just links that are being ruled out, I'm pretty sure we can still print a menu and just enter 1, 2, 3, etc...

I'll test that now.

**Comment by ThePix** [comment]
I would (obviously) prefer my library, which offers the player a menu, with each topic being an object Topics can be added and removed dynamically as the game progresses.
https://github.com/ThePix/quest/wiki/Conversations:-Library

**Comment by hegemonkhan** [comment]
I've still got a long ways to go to get to your level Pixie, this was just my code on doing custom menu display and choices, using quick number inputs.

I see you got this handled already, along with being combined with dialogue system in your dialogue library (didn't know this type of stuff was handled in your conversation library)

(I need to get around to studying/looking at other peoples, especially/mainly yours, again, to learn more)

**Comment by rheadkid** [comment]
Started up an issue, probably should have just asked here...

See the latest issue concerning libraries, though.

**Comment by XanMag** [comment]
I am all for the original ask/tell.  I'm not a fan of menus.  I think it gives too large a hints to the topics that can be discussed.  That being said, I'm also pretty flexible, so if the choice is overwhelmingly "Let's use menus", I can do that too.

**Comment by DarkLizerd** [comment]
One question, hopefully, this will simplify talking...
">talk to farmer
(now the program in in talk mode, you cannot move or do anything else, then all talk can be done with just one word instead of:
">ask farmer about hoes
">ask farmer about pouch
">ask farmer about rats
">ask farmer about (whatever)
)
Sort of like: player.IsTalking=true
then its like:
">talk farmer
(player.IsTalking=true)
">hoes
">pouch
">north
Sorry, never been there myself.
">bye 
You stopped talking to the farmer and get ready to move out.
player.IsTalking=false
">north
You go north to somewhere.
Altho, you should still have access to 'inv' and maybe a few other commands.

**Comment by rheadkid** [comment]
Infocom does that in Bureaucracy.

The indicator switches from `>` to ~>>~ while in talking mode, but we can just make the input text field box display 'YOU ARE IN CONVERSATION MODE!!!' or something.

For the record: I found conversation more VERY unpleasant.

For one more record: the point of Bureaucracy: Douglas Adams and Steve Meretzky are PISSING THE PLAYER OFF AT ALMOST EVERY TURN, but the pay-off is HILARIOUS!

---

I just added the topics string list to the farmer, so now there are choices, which have hyperlinks, even though hyperlinks are off... I may be overlooking something else though.
